file.go

```diff
@@ -485,6 +488,48 @@ func (c *fileColumnChunk) NumValues() int64 {
 	return c.chunk.MetaData.NumValues
 }
 
+func (c *fileColumnChunk) readColumnIndex() {
+	if c.columnIndex != nil {
+		return
+	}
+	chunkMeta := c.file.metadata.RowGroups[c.rowGroup.Ordinal].Columns[c.Column()]
+	offset, length := chunkMeta.ColumnIndexOffset, chunkMeta.ColumnIndexLength
+	if length == 0 {
+		return
+	}
+
+	indexData := make([]byte, int(length))
+	var columnIndex format.ColumnIndex
+	if _, err := readAt(c.file.reader, indexData, offset); err != nil {
+		return
```

Comment:
We might want to handle this case more gracefully, but the method does not return an error right now and we would have to make a breaking change. Alternatively we could also panic to indicate a malformed file.

Reply:
I would favor an API change

Reply:
agreed with frederic. obviously we shouldn't rewrite the API, but we should feel comfortable adding an error return to a method when it makes sense.

Reply:
Makes sense, updated!

Reply:
Gentle ping @joe-elliott @brancz